test/performance/specs/post-editor.spec.js

```diff
@@ -0,0 +1,390 @@
+	listViewOpen: [],
+	inserterOpen: [],
+	inserterHover: [],
+	inserterSearch: [],
+};
+
+test.describe( 'Post Editor Performance', () => {
+	const traceFilePath = getTraceFilePath();
+
+	test.afterAll( async () => {
+		saveResultsFile( __filename, results );
+		deleteFile( traceFilePath );
+	} );
+
+	test.beforeEach( async ( { admin, page } ) => {
+		await admin.createNewPost();
+		// Disable auto-save to avoid impacting the metrics.
+		await page.evaluate( () => {
+			window.wp.data.dispatch( 'core/editor' ).updateEditorSettings( {
+				autosaveInterval: 100000000000,
+				localAutosaveInterval: 100000000000,
+			} );
+		} );
+	} );
+
+	test( 'Loading', async ( { browser, page } ) => {
+		await loadBlocksFromHtml(
+			page,
+			path.join( process.env.ASSETS_PATH, 'large-post.html' )
+		);
+
+		await page
+			.getByRole( 'button', { name: 'Save draft' } )
+			.click( { timeout: 60_000 } );
+		await expect(
+			page.getByRole( 'button', { name: 'Saved' } )
+		).toBeDisabled();
+
+		const draftURL = page.url();
+
+		// Number of sample measurements to take.
+		const samples = 5;
+		// Number of throwaway measurements to perform before recording samples.
+		// Having at least one helps ensure that caching quirks don't manifest in
+		// the results.
+		const throwaway = 1;
+
+		let i = throwaway + samples;
+		while ( i-- ) {
+			const testPage = await browser.newPage();
+
+			await testPage.goto( draftURL );
+			await testPage
+				.frameLocator( 'iframe[name="editor-canvas"]' )
+				.locator( '.wp-block' )
+				.first()
+				.waitFor( {
+					timeout: 120_000,
+				} );
+
+			if ( i < samples ) {
+				const {
+					serverResponse,
+					firstPaint,
+					domContentLoaded,
+					loaded,
+					firstContentfulPaint,
+					firstBlock,
+				} = await getLoadingDurations( testPage );
+
+				results.serverResponse.push( serverResponse );
+				results.firstPaint.push( firstPaint );
+				results.domContentLoaded.push( domContentLoaded );
+				results.loaded.push( loaded );
+				results.firstContentfulPaint.push( firstContentfulPaint );
+				results.firstBlock.push( firstBlock );
+			}
+
+			await testPage.close();
+		}
+	} );
+
+	test( 'Typing', async ( { browser, page, editor } ) => {
+		await loadBlocksFromHtml(
+			page,
+			path.join( process.env.ASSETS_PATH, 'large-post.html' )
+		);
+		await editor.insertBlock( { name: 'core/paragraph' } );
+
+		await browser.startTracing( page, {
+			path: traceFilePath,
+			screenshots: false,
+			categories: [ 'devtools.timeline' ],
+		} );
+
+		let i = 20;
+		while ( i-- ) {
+			// Wait for the browser to be idle before starting the monitoring.
+			// The timeout should be big enough to allow all async tasks tor run.
+			// And also to allow Rich Text to mark the change as persistent.
+			// eslint-disable-next-line no-restricted-syntax
+			await page.waitForTimeout( 2000 );
+			await page.keyboard.type( 'x' );
+		}
+
+		await browser.stopTracing();
+		const traceResults = JSON.parse( readFile( traceFilePath ) );
```

Comment:
If we're only saving the trace for this file then we could probably just use the buffer returned from [`stopTracing()`](https://playwright.dev/docs/api/class-browser#browser-stop-tracing). Something like (not tested)...

```js
const buffer = await browser.stopTracing();
const traceResults = JSON.parse( buffer.toString( 'utf8' ) );
```

Reply:
The trace file is saved to disk and attached to the failure artifacts zip if the test fails somewhere in the middle. I'm not sure how useful that file is debugging-wise, but we should probably decide on it in a separate PR. /cc @dmsnell

Reply:
agree on not changing it here. it's only useful when things break; and when things break is when it's most valuable.

Reply:
We can decide it in a separate issue/PR but maybe a Playwright trace file is more helpful than a Chromium trace file?

Reply:
Missed this question, apologies. We're using Chromium trace to calculate e.g. typing events durations. Isn't Playwright trace strictly for using with TraceViewer/debugging?